Source/Core/DolphinQt2/FIFOPlayerWindow.cpp

```diff
@@ -0,0 +1,359 @@
+}
+
+void FIFOPlayerWindow::UpdateInfo()
+{
+  if (FifoPlayer::GetInstance().GetFile() != nullptr && Core::IsRunning())
+  {
+    FifoDataFile* file = FifoPlayer::GetInstance().GetFile();
+    m_info_label->setText(
+        tr("%1 frame(s)\n%2 object(s)\nCurrent Frame: %3")
+            .arg(QString::number(file->GetFrameCount()),
+                 QString::number(FifoPlayer::GetInstance().GetFrameObjectCount()),
+                 QString::number(FifoPlayer::GetInstance().GetCurrentFrameNum())));
+    return;
+  }
+
+  if (FifoRecorder::GetInstance().IsRecording())
+  {
+    m_info_label->setText(tr("Recording..."));
+    return;
+  }
+
+  m_info_label->setText(tr("No file loaded / recorded."));
+}
+
+void FIFOPlayerWindow::OnFIFOLoaded()
+{
+  FifoDataFile* file = FifoPlayer::GetInstance().GetFile();
+
+  auto object_count = FifoPlayer::GetInstance().GetFrameObjectCount();
+  auto frame_count = file->GetFrameCount();
+
+  m_frame_range_to->setMaximum(frame_count);
+  m_frame_range_to->setValue(frame_count);
+  m_object_range_to->setMaximum(FifoPlayer::GetInstance().GetFrameObjectCount());
+  m_object_range_to->setValue(object_count);
+
+  UpdateInfo();
+  UpdateLimits();
+  UpdateControls(Core::IsRunning());
+}
+
+void FIFOPlayerWindow::OnFrameFromChanged(int value)
+{
+  FifoPlayer& player = FifoPlayer::GetInstance();
+
+  player.SetFrameRangeStart(value);
+
+  UpdateLimits();
+}
+
+void FIFOPlayerWindow::OnFrameToChanged(int value)
+{
+  FifoPlayer& player = FifoPlayer::GetInstance();
+  player.SetFrameRangeEnd(value);
+
+  UpdateLimits();
+}
+
+void FIFOPlayerWindow::OnObjectFromChanged(int value)
+{
+  FifoPlayer::GetInstance().SetObjectRangeStart(value);
+
+  UpdateLimits();
+}
+
+void FIFOPlayerWindow::OnObjectToChanged(int value)
+{
+  FifoPlayer::GetInstance().SetObjectRangeEnd(value);
+
+  UpdateLimits();
+}
+
+void FIFOPlayerWindow::OnEarlyMemoryUpdatesChanged(bool enabled)
+{
+  FifoPlayer::GetInstance().SetEarlyMemoryUpdates(enabled);
+}
+
+void FIFOPlayerWindow::UpdateLimits()
+{
+  m_frame_range_from->setMaximum(m_frame_range_to->value() - 1);
+  m_frame_range_to->setMinimum(m_frame_range_from->value() + 1);
+
+  m_object_range_from->setMaximum(m_object_range_to->value() - 1);
+  m_object_range_to->setMinimum(m_object_range_from->value() + 1);
+}
+
+void FIFOPlayerWindow::UpdateControls(bool running)
+{
+  bool is_playing = FifoPlayer::GetInstance().GetFile() != nullptr && running;
+
+  m_frame_range_from->setEnabled(is_playing);
+  m_frame_range_from_label->setEnabled(is_playing);
+  m_frame_range_to->setEnabled(is_playing);
+  m_frame_range_to_label->setEnabled(is_playing);
+  m_object_range_from->setEnabled(is_playing);
+  m_object_range_from_label->setEnabled(is_playing);
+  m_object_range_to->setEnabled(is_playing);
+  m_object_range_to_label->setEnabled(is_playing);
+
+  m_early_memory_updates->setEnabled(is_playing);
+
+  bool enable_count = !is_playing && !FifoRecorder::GetInstance().IsRecording();
```

Comment:
`enable_count` could refer to a lot of things... how about `enable_frame_record_count`?